src/background/actions/injection-action-creator.ts

```diff
@@ -19,11 +23,22 @@ export class InjectionActionCreator {
             Messages.Visualizations.State.InjectionCompleted,
             this.injectionCompleted,
         );
+        this.interpreter.registerTypeToPayloadCallback(
+            Messages.Visualizations.State.InjectionFailed,
+            this.injectionFailed,
+        );
     }
 
     private injectionStarted = async (): Promise<void> =>
         await this.injectionActions.injectionStarted.invoke(null);
 
     private injectionCompleted = async (): Promise<void> =>
         await this.injectionActions.injectionCompleted.invoke(null);
+
+    private injectionFailed = async (payload: InjectionFailedPayload): Promise<void> => {
+        if (payload.injectionFailed) {
```

Comment:
The naming of the message vs the `injectionFailed` property makes this consuming code a bit confusing to read - the message is named `InjectionFailed`, but it only sometimes indicated that injection failed?

Maybe instead something like `failedPermanently`, or invert it as `shouldRetry`/`willRetry`?